syscall.go

```diff
@@ -111,6 +114,62 @@ func unmountShareDir() error {
 	return os.RemoveAll(containerMountDest)
 }
 
+func waitForBlockDevice(deviceName string) (err error) {
+	devicePath := filepath.Join(devPath, deviceName)
+
+	if _, err := os.Stat(devicePath); err == nil {
+		return nil
+	}
+
+	u := goudev.Udev{}
+
+	// Create a monitor listening on a NetLink socket
+	monitor := u.NewMonitorFromNetlink("udev")
+
+	// Add filter to watch for just block devices
+	if err := monitor.FilterAddMatchSubsystemDevtype("block", "disk"); err != nil {
+		return err
+	}
+
+	// Create done signal channel for signalling epoll loop on the monitor socket.
+	done := make(chan struct{})
+
+	// Create channel to signal when desired udev event has been received.
+	doneListening := make(chan bool)
+
+	// Start monitor goroutine
+	ch, _ := monitor.DeviceChan(done)
+
+	go func() {
+		agentLog.Infof("Started listening for udev events for block device hotplug")
+
+		// Check if the device already exists
+		if _, err := os.Stat(devicePath); err == nil {
+			agentLog.Infof("Device %s already hotplugged, quit listening", deviceName)
+			close(done)
+			close(doneListening)
```

Comment:
Please put a return at the end of this statement, this would allow avoiding the `else` statement which is not needed.